### ISSUE TITLE
Feat Request: include basename of current file when running git commnads

### ISSUE THREAD

**Issue by mrsdizzie** [linrongbin16/gitlinker.nvim#231]
I have a base folder "projects" that contains many different git repos. If I open the "projects" folder in neovim and navigate to a file inside of a repo, gitlink doesn't work because it doesn't pass a specific directory to `run_cmd` when running various git commands. `run_cmd` will just use the vim.cwd value which will be the "projects" folder that doesn't contain any relevant git information and gitlink will say something like "not a git repository (or any of the parents)".

I believe if it passed the directory the current file was in when using `run_cmd`, git itself would do the right thing and return the correct information.

Thanks!

**Comment by linrongbin16** [comment]
hi @mrsdizzie ,

is it a git worktree, or bare repository?

can you give me an example that reproduce this issue?

**Comment by mrsdizzie** [comment]
Just normal git clones. A example to reproduce:

```sh
~ $ mkdir projects
~ $ cd projects
~/projects $ git clone https://github.com/linrongbin16/gitlinker.nvim
Cloning into 'gitlinker.nvim'...
remote: Enumerating objects: 2015, done.
remote: Counting objects: 100% (551/551), done.
remote: Compressing objects: 100% (237/237), done.
remote: Total 2015 (delta 439), reused 347 (delta 304), pack-reused 1464
Receiving objects: 100% (2015/2015), 564.20 KiB | 11.28 MiB/s, done.
Resolving deltas: 100% (1159/1159), done.
~/projects $ nvim .
```
Now neovim is open with `~/projects` as the root dir. if I navigate to `gitlinker.nvim` and try to get a URL it says it is not a git repo. I'm guessing if it passed the basename of the open file [ `~/projects/gitlinker.nvim` in this example] as the cwd to `run_cmd` then it would work as git returns the right thing when run within the `~/projects/gitlinker.nvim` folder.

<img width="1512" alt="Screenshot 2024-03-22 at 18 29 23" src="https://github.com/linrongbin16/gitlinker.nvim/assets/1669571/919881e6-feec-4857-9964-2f4e0928296d">

The use case is having many separate git repositories that are related and wanting neovim open in a root folder that has access to all of them at once.

**Comment by linrongbin16** [comment]
thanks, it looks more like a bug, I will fix it.

**Comment by linrongbin16** [comment]
hi @mrsdizzie ,

This issue is fixed in #232 , please pull master branch and have a try!

**Comment by mrsdizzie** [comment]
Amazing, thank you!! Tested and seems to work well.
